### DISCUSSION TITLE
NextJs doesn’t set locale for Link in components from node_modules

### DISCUSSION THREAD

**Discussion by andrey-kudinov** [vercel/next.js#34446] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Describe the Bug

We use Storybook for custom components.
When we use this components (from `node_modules`) in main site, nextjs doesn’t set locale for `Link` in our custom components.
Locale appears after rerendering component.

Obviously i18n works before components load from `node_modules`.

### Expected Behavior

We expect that locale in `Link` will be without rerendering.

### To Reproduce

You need using some components with `Link` from `node_modules`.

**Comment by balazsorban44** [comment]
Hi, Storybook recently created a zero-config addon for Next.js, check it out here: https://storybook.js.org/addons/storybook-addon-next

Otherwise, I'm not sure if `Link` is supposed to be supported when coming from `node_modules` right now. In that case, this is a feature request rather than a bug.

**Reply by andrey-kudinov** [reply]
Idn, `locale` appears after force unmount/mount
I think `locale` in `Link` should appear at the first render

**Reply by fasani-tx** [reply]
@andrey-kudinov Do you think the only solution today then would be to pass the `next/link` to a UI lib component via a prop?

**Reply by andrey-kudinov** [reply]
Hi @fasani-tx !
I used that solution
I didn't think of anything better
So now I get extra re-render

<img width="602" alt="image" src="https://user-images.githubusercontent.com/64890665/185431030-c394bc7f-694e-464a-86b0-09ab385fac0a.png">

**Comment by wuifdesign** [comment]
having the same problem. even if setting the locale 

for an url "/de/test"

```
const { locale } = useRouter()
// locale === "de"
  
<Link href="/demo" locale={locale}>
  {item.title}
</Link>
```

the link loaded from node_modules still is "/demo" and not "/de/demo" throwing following warning:

```Prop `href` did not match. Server: "/demo" Client: "/de/demo"```

i use this as i have a separate package for my menu component

a workaround would be to use https://www.npmjs.com/package/next-transpile-modules, than it will work fine. but i would like to remove it.

**Reply by fasani-tx** [reply]
@wuifdesign did you solve it? I think I have the same issue as half of my recently inherited code base works fine and is using next-transpile-modules the other half comes from a Lerna lib with a link and I have no locale on first render which happens to be server side also.
